### PR TITLE
Adds military helipad location

### PR DESCRIPTION
**Pull request by CountAlex**
#### Summary
SUMMARY: Content "Adds new mapgen location - military helipad"

#### Purpose of change
New location.

#### Describe the solution
This is small military helicopter pad outpost. Nothing like full-scale helicopter base (those are rare) or something like that.
Just a small military location with 3 pads, hangar, barracks and some guarding turrets.

#### Testing
Main level - pads, offices, cantina, warehouse, parking lot and entrance.
main

2nd level - barracks and roofs.
roof

#### Additional context
I will add screenshots so people might have a better look.
This also adds outdoor version of yellow concrete so pads can be accented.

### CODE REVIEW

**Comment by curstwist** [comment]
screenshots please.

**Comment by Amoebka** [comment]
This should probably have a single-frame crane vehicle instead of the guaranteed lifter mech. Those are rare prototypes, not standart issue equipment for small military installations.

**Comment by CountAlex** [comment]
> screenshots please.
Let me weed out errors, and I will throw them in.

> This should probably have a single-frame crane vehicle instead of the guaranteed lifter mech. Those are rare prototypes, not standart issue equipment for small military installations.
Then I will figure something out. Forklift car might be a good solution.

File: data/json/mapgen_palettes/helipad.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "-": "t_floor",
+      "w": "t_window",
+      "G": "t_chaingate_l",
+      "|": "t_concrete_wall",
+      "F": "t_fence_barbed",
+      "!": "t_guardrail",
+      "p": "t_pavement",
+      "m": "t_door_metal_c",
+      "y": "t_pavement_y",
+      ".": "t_grass",
```
Comment (I-am-Erk) [inline data/json/mapgen_palettes/helipad.json]:
```suggestion
      ".": "t_region_groundcover_urban",
```

File: data/json/mapgen_palettes/helipad.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      "T": "t_grass",
```
Comment (I-am-Erk) [inline data/json/mapgen_palettes/helipad.json]:
```suggestion
      "T": "t_region_groundcover_urban",
```

File: data/json/mapgen/helipad.json [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        "FTFFFFFFFFFFFFFFTFFFFFFFFFFFFFFTFFFFFFFFFFFFFFTF",
+        "FFF............FFF............FFF............FFF"
+      ],
+      "palettes": [ "helipad_palette" ],
+      "toilets": { "&": {  } },
+      "gaspumps": { "$": {  } },
+      "items": {
+        "B": {
+          "item": {
+            "subtype": "distribution",
```
Comment (I-am-Erk) [inline data/json/mapgen/helipad.json]: This seems like something that could be defined in an itemgroup, perhaps in supplies.json. I can see other people having use for drums of gasoline and motor oil

File: data/json/mapgen/helipad.json [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        "d": { "item": "SUS_office_desk" },
+        "f": { "item": "SUS_office_filing_cabinet" },
+        "4": { "item": "military_standard_shotguns", "chance": 50 },
+        "#": { "item": "SUS_utensils", "chance": 25, "repeat": [ 1, 3 ] },
+        "t": { "item": "SUS_dishes", "chance": 50, "repeat": [ 1, 2 ] },
+        "[": { "item": "SUS_fridge" },
+        "O": { "item": "SUS_oven" },
+        "H": { "item": "SUS_spice_collection" }
+      },
+      "place_monsters": [
```
Comment (I-am-Erk) [inline data/json/mapgen/helipad.json]: I don't know the groups well but I'd assume the regular, non-weak military groups should also have a chance of spawning here if it's a major outpost with a helipad.

File: data/json/mapgen/helipad.json (hunk 3 above)
Comment (CountAlex) [reply data/json/mapgen/helipad.json]: I've added it as you suggested. I will give it some test about spawned amounts - I am not sure how charges work with fuels exactly. Besides I'd like to have some opinion on how much fuel spawned barrels should have. 
I also think motor oil might need to have different container than plastic jerrycan.

File: data/json/mapgen/helipad.json (hunk 4 above)
Comment (CountAlex) [reply data/json/mapgen/helipad.json]: Tweaked that a bit. 
There are three military monstergroups, GROUP_MIL_WEAK, GROUP_MIL_STRONG and GROUP_MAYBE_MIL. First two have different sizes and stronger Zs in latter case, third one also adds military robots to the mix.

**Comment by CountAlex** [comment]
Cleaned up some things and added screenshots. 
Also moved copters spawning points a bit and decreased chance of them spawning since they spawn way too intact without making it hardcoded as it works for map extras.

**Comment by Amoebka** [comment]
10 turrets is 16.000 rounds of 5.56 a player can loot from one location.
Just saying.

**Comment by I-am-Erk** [comment]
The ammo loot isn't really a consideration when designing a map. That's a problem with calculating turret age, not mapgen.

**Comment by Amoebka** [comment]
Yes, but mapgen can still take it into account if there are no downsides.

10 turrets don't exactly guard the location any better than 4. Gameplay-wise turrets are gear checks, and a player that can deal with 4 turrets can just as easily deal with 10, the only difference being the ammo reward. Story-wise, this is a small refueling base deep inside home territory, it doesn't need multiple machinegun nests with overlapping firing arcs. A few turrets would do just fine to deter theoretical terrorists and random idiots.

**Comment by CountAlex** [comment]
Changing amount of turrets will take me two minutes while this is still at PR stage. Just tell me if I need to.

**Comment by Amoebka** [comment]
I would say leave just the 4 turrets in the corners. Their ranges should be high enough to cover the entire perimeter anyway.

**Comment by CountAlex** [comment]
Yeah, I guess it makes sense. Missile silo also has only corner ones (and couple near entrance) while being way more significant military object.

**Pull request merged**